tests/test_utils/test_simpletestcase.py

```diff
@@ -0,0 +1,90 @@
+import unittest
+from io import StringIO
+from unittest import mock
+from unittest.suite import _DebugResult
+
+from django.test import SimpleTestCase
+
+
+class ErrorTestCase(SimpleTestCase):
+    def raising_test(self):
+        self._pre_setup.assert_called_once_with()
+        raise Exception('Should bubble up when using debug()')
+
+    def simple_debug_test(self):
+        self._pre_setup.assert_called_once_with()
+
+    @unittest.skip('Testing debug skip conditions')
+    def skipped_test(self):
+        pass
+
+
+@mock.patch.object(ErrorTestCase, '_post_teardown')
+@mock.patch.object(ErrorTestCase, '_pre_setup')
+class DebugInvocationTests(SimpleTestCase):
+    def get_runner(self):
+        return unittest.TextTestRunner(stream=StringIO())
+
+    def test_post_teardown_called(self, _pre_setup, _post_teardown):
+        """Simple test run: catches errors and runs cleanup."""
+        test_suite = unittest.TestSuite()
+        case = ErrorTestCase('raising_test')
+        test_suite.addTest(case)
+        test_suite.run(self.get_runner()._makeResult())
+        _pre_setup.assert_called_once_with()
+        _post_teardown.assert_called_once_with()
+
+    def test_debug_bubbles_exception(self, _pre_setup, _post_teardown):
+        """debug() should bubble up exceptions before any cleanup."""
```

Comment:
remove "should", e.g. "debug() bubbles up exceptions before cleanup."

Reply:
addressed in 3bf9e42